runtime/src/main/java/org/corfudb/runtime/LogReplicationLogicalGroupClient.java

```diff
@@ -0,0 +1,357 @@
+            tempSourceMetadataTable = corfuStore.openTable(CORFU_SYSTEM_NAMESPACE, LR_SOURCE_METADATA_TABLE_NAME, ClientDestinationInfoKey.class,
+                    DestinationInfoVal.class, null, TableOptions.fromProtoSchema(DestinationInfoVal.class));
+        }
+        this.sourceMetadataTable = tempSourceMetadataTable;
+
+        register();
+    }
+
+    /**
+     * Adds client and model to LogReplicationRegistrationTable
+     */
+    private void register() {
+        try {
+            IRetry.build(ExponentialBackoffRetry.class, () -> {
+                try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+                    if (txn.isExists(LR_REGISTRY_TABLE_NAME, clientKey)) {
+                        log.warn("Client already registered \nclientKey: {} \nclientInfo: {}",
+                                clientKey, txn.getRecord(replicationRegistrationTable, clientKey).getPayload());
+                    } else {
+                        txn.putRecord(replicationRegistrationTable, clientKey, clientInfo, null);
+                        txn.commit();
+                    }
+                } catch (TransactionAbortedException tae) {
+                    log.error(String.format("Unable to register client %s, Error: ", clientName), tae);
+                    throw new RetryNeededException();
+                }
+                return null;
+            }).run();
+        } catch (InterruptedException e) {
+            log.error("Client registration failed");
+            throw new UnrecoverableCorfuInterruptedError(e);
+        }
+    }
+
+    /**
+     * Add destination to LogReplicationSourceMetadataTable for monitoring
+     *
+     * @param logicalGroup String representative of the group of tables to be replicated
+     * @param destination Cluster which the logicalGroup should be replicated to
+     */
+    public void addDestination(String logicalGroup, String destination) {
+        Preconditions.checkArgument(isValid(logicalGroup), "logicalGroup is null or empty");
+        Preconditions.checkArgument(isValid(destination), "destination is null or empty");
+        try {
+            IRetry.build(ExponentialBackoffRetry.class, () -> {
+                try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+                    ClientDestinationInfoKey clientInfoKey = ClientDestinationInfoKey.newBuilder()
+                            .setClientName(clientName)
+                            .setModel(LogReplication.ReplicationModel.valueOf(modelEnum))
+                            .setGroupName(logicalGroup)
+                            .build();
+
+                    DestinationInfoVal clientDestinations;
+
+                    if (txn.isExists(LR_SOURCE_METADATA_TABLE_NAME, clientInfoKey)) {
+                        clientDestinations = txn.getRecord(sourceMetadataTable, clientInfoKey).getPayload();
+
+                        if (clientDestinations.getDestinationIdsList().contains(destination)) {
```

Comment:
IMO, this should not be treated as an error.  Just logging a debug statement and returning is sufficient